### DISCUSSION TITLE
Hide Spaces without Windows

### DISCUSSION THREAD

**Discussion by TaylorFinklea** [FelixKratz/SketchyBar#386] [Q&A] [answered]
Hi there,

I am sure there's a way to do this, but can't figure out how. I would like to only show spaces that have a window on them, so like in the below screenshot, spaces 5-10 would not show up unless I went there an opened another window. I'm using the spaces.sh Item and Plugin from [**Here**](https://github.com/FelixKratz/dotfiles)

![image](https://github.com/FelixKratz/SketchyBar/assets/6810234/6b167d34-8566-47fa-8ab2-6b6f39a27da3)

**Comment by FelixKratz** [comment, marked as answer]
Hey there,

sure this is possible to do. The logic where I look up the icons for the applications is the ideal place to put this additional logic. It is in the file `plugins/yabai.sh` in my config.

Basically, you want to set the space to not draw if it has no apps on them, i.e. in the function where the app icons are looked up (`windows_on_spaces` in my config) you can simply add some lines in the appropriate places, i.e.
`args+=(--set space.$space drawing=on)` and `args+=(--set space.$space drawing=off)` such that you arrive at the modified `windows_on_spaces` function which looks like this:
```bash
windows_on_spaces () {
  CURRENT_SPACES="$(yabai -m query --displays | jq -r '.[].spaces | @sh')"

  args=(--set spaces_bracket drawing=off
        --set '/space\..*/' background.drawing=on
        --animate sin 10)

  while read -r line
  do
    for space in $line
    do
      icon_strip=" "
      apps=$(yabai -m query --windows --space $space | jq -r ".[].app")
      if [ "$apps" != "" ]; then
        args+=(--set space.$space drawing=on)
        while IFS= read -r app; do
          icon_strip+=" $($CONFIG_DIR/plugins/icon_map.sh "$app")"
        done <<< "$apps"
      else
        args+=(--set space.$space drawing=off)
      fi
      args+=(--set space.$space label="$icon_strip" label.drawing=on)
    done
  done <<< "$CURRENT_SPACES"

  sketchybar -m "${args[@]}"
}
```

**Reply by TaylorFinklea** [reply]
This is the best, thank you for the complete reply and a description of how it works. Looks so much better now!
